src/360-core/browzine-360-core-adapter.js

```diff
@@ -6,31 +11,28 @@ browzine.serSol360Core = (function() {
     return url.replace("api.thirdiron.com", "public-api.thirdiron.com");
   };
 
-  function getQueryVariable(url, key) {
-    var query = url.split("?")[1];
-    var parameters = query.split("&");
-    var value = "";
-
-    for(var i = 0; i < parameters.length; i++) {
-      var pair = parameters[i].split("=");
+  function getIssn(title) {
+    if(title.identifiers) {
+      var issnIdentifier = title.identifiers.filter(function(identifier) {
+        if(identifier.type && identifier.value) {
+          return identifier.type.toLowerCase() === "issn" && identifier.value.length > 0;
+        }
+      }).pop();
 
-      if(pair[0] == key) {
-        value = pair[1];
-        return value;
+      if(issnIdentifier) {
+        return encodeURIComponent(issnIdentifier.value);
       }
-    }
 
-    return value;
-  };
-
-  function getIssn(title) {
-    var issn = "";
+      var eissnIdentifier = title.identifiers.filter(function(identifier) {
+        if(identifier.type && identifier.value) {
+          return identifier.type.toLowerCase() === "eissn" && identifier.value.length > 0;
+        }
+      }).pop();
 
-    if(title.syndeticsImageUrl) {
-      issn = getQueryVariable(title.syndeticsImageUrl, "issn");
+      if(eissnIdentifier) {
+        return encodeURIComponent(eissnIdentifier.value);
+      }
```

Comment:
If we don't have an EISSN, we'll end up returning `undefined` here, is that alright? Or do we not expect to get in that case?

Reply:
Without an EISSN, we won't hit that return block at all. Added some more test cases to validate the same. Good question, Thanks!